src/utils/validation.ts

```diff
@@ -31,3 +35,48 @@ export const isTestnet = (): boolean => {
 
   return network === 'mainnet' ? false : true;
 };
+
+export const getOnchainMetadataVersion = (onchainMetadata: Asset['onchain_metadata']): 1 | 2 => {
+  if (!onchainMetadata?.version) {
+    return 1;
+  }
+
+  if (Number(onchainMetadata.version) === 2) {
+    return 2;
+  }
+
+  return 1;
+};
+
+export const getOnchainMetadata = (
+  onchainMetadata: Asset['onchain_metadata'],
+  assetName: Asset['asset_name'],
+  policyId: Asset['policy_id'],
+) => {
+  if (!onchainMetadata || !assetName) return null;
```

Comment:
couldn't assetName be an empty string? that's still a valid token (composed of only policy id) that could have some onchain metadata?

Reply:
Can be empty string but null?

Reply:
https://github.com/blockfrost/blockfrost-backend-ryo/pull/18/commits/96b81f7edc81977932209e0e8bccd290aa272bb5#diff-f9d8e0bc7f251fefd3b4f00fe043bd526c470ae6131c26b40f5ceb8f4264e4f1R577